ddp-workspace/projects/ddp-atcp/src/app/sdk/services/currentActivity.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { ActivityServiceAgent, ActivityForm } from 'ddp-sdk';
+import { filter, tap } from 'rxjs/operators';
+import { ActivityProgressCalculationService } from './activityProgressCalculation.service';
+import { ActivityCodes } from '../constants/activityCodes';
+
+@Injectable()
+export class CurrentActivityService {
+  private activity = new BehaviorSubject(null);
+
+  constructor(private activityServiceAgent: ActivityServiceAgent,
+              private activityProgressCalculationService: ActivityProgressCalculationService) {
+    this.activity
+      .pipe(filter(x => x !== null))
+      .subscribe(x => this.activityProgressCalculationService.setProgress(x));
+  }
+
+  public getActivity(studyGuid: Observable<string | null>,
+                     activityGuid: Observable<string | null>): Observable<ActivityForm> {
+    return this.activityServiceAgent
+      .getActivity(studyGuid, activityGuid)
+      .pipe(tap(x => this.saveCurrentActivity(x)));
+  }
+
+  public updateActivitySection(studyGuid: string,
+                               activityGuid: string,
+                               activity: ActivityForm,
+                               sectionIndex: number): void {
+    if (activity.activityCode === ActivityCodes.MEDICAL_HISTORY) {
+      this.activityProgressCalculationService.updateProgress(activity, sectionIndex);
+      this.activityServiceAgent
+        .saveLastVisitedActivitySection(studyGuid, activityGuid, sectionIndex)
+        .subscribe();
```

Comment:
Consider modifying to something like:
```suggestion
      this.activityServiceAgent
        .saveLastVisitedActivitySection(studyGuid, activityGuid, sectionIndex)
        .pipe(take(1))
        .subscribe();
```
to ensure the subscription is immediately disposed of.

Reply:
done